### PR TITLE
Action link save in progress intro

### PR DESCRIPTION
**Pull request by jaredcunha**
## Original issue(s)
department-of-veterans-affairs/va.gov-team#42237

# Expected behavior
The "Start without signing in" link should use a link element so that users are able to open a new tab. Screen reader users should have the element read as a link.

## Current behavior
The "Start without signing in" link is written as a button which is materially dishonest. Users may become frustrated that they are unable to open a new tab. Screen reader users may be confused that the text seems more like a link despite being announced as a button.

## Your fix
- Link uses link element (technically `Link` component)
- Add missing proptypes
- Migrate loading indicator to `va-loading-indicator`

## How has this been tested?
Describe the process you used to test that your fix is resulting in the expected behavior. Feel free to use a bullet list.
- Visit `localhost:3001//burials-and-memorials/application/530/introduction/
- Right click on "Start your application without signing in"
- Confirm you can open in a new tab 

## Screenshots
Screen Shot 2022-06-28 at 8 09 23 PM

## Acceptance criteria
- [x] No changes in behavior
- [x] Tests continue to pass
- [x] Link uses link element  

## Definition of done
- [x] Link to original issue provided

### CODE REVIEW

**Review by holdenhinkle — APPROVED** [review]

**Review by micahchiang — APPROVED** [review]

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx [hunk 1]
```diff
@@ -209,14 +208,15 @@ class SaveInProgressIntro extends React.Component {
           {unauthStartButton}
           {!this.props.hideUnauthedStartLink && (
             <p>
-              <button
-                className="va-button-link schemaform-start-button"
+              <Link
                 onClick={this.goToBeginning}
```
Comment (bkjohnson) [inline src/platform/forms/save-in-progress/SaveInProgressIntro.jsx:212]: With the change to `<Link>`, it looks like the `router.push` is [already being handled](https://github.com/remix-run/react-router/blob/850de933444d260bfc5460135d308f9d74b52c97/modules/Link.js#L95). Do we still need to manually be doing that?

https://github.com/department-of-veterans-affairs/vets-website/blob/61fec575eb0940059e0b56efd93bfd41c9851b00/src/platform/forms/save-in-progress/SaveInProgressIntro.jsx#L302-L305

Comment (jaredcunha) [reply src/platform/forms/save-in-progress/SaveInProgressIntro.jsx:212]: Yeah - I wasn't sure if I should leave that in or not either. I suppose if we got rid of that, should probably rename this to `recordNoLoginStartEvent` or something. What do you think?

**Review by OptionSelect — APPROVED** [review]
LGTM! 🚀

**Review by lzapson — APPROVED** [review]
hca related changes looks good to me

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx [hunk 2]
```diff
@@ -388,41 +389,48 @@ class SaveInProgressIntro extends React.Component {
...
+  pathname: PropTypes.string,
+  prefillEnabled: PropTypes.bool,
+  prefillTransformer: PropTypes.func,
+  renderSignInMessage: PropTypes.func,
+  resumeOnly: PropTypes.bool,
+  retentionPeriod: PropTypes.string,
+  returnUrl: PropTypes.string,
+  router: PropTypes.shape({
+    push: PropTypes.func,
+  }),
```
Comment (bkjohnson) [inline src/platform/forms/save-in-progress/SaveInProgressIntro.jsx:426]: I don't think we need this anymore.

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx (hunk 1 above)
Comment (bkjohnson) [reply src/platform/forms/save-in-progress/SaveInProgressIntro.jsx]: The `handleClick` that you've got now seems fine to me.

**Review by bkjohnson — APPROVED** [review]

**Review by adamwhitlock1 — APPROVED** [review]

**Review by jerekshoe — APPROVED** [review]

**Review by dan-brooking — APPROVED** [review]

**Pull request merged**